packages/tools/devtools/devtools-core/src/ContainerRuntimeDevtools.ts

```diff
@@ -99,3 +107,59 @@ export class ContainerRuntimeDevtools extends BaseDevtools<DecomposedContainer>
 		return false;
 	}
 }
+
+/**
+ * Implementation of {@link DecomposedContainer} that wraps an {@link @fluidframework/container-runtime-definitions/internal#IContainerRuntime}.
+ * This class provides a bridge between {@link @fluidframework/container-runtime-definitions/internal#IContainerRuntime} and the devtools system by exposing runtime properties and events.
+ */
+export class DecomposedContainerForContainerRuntime
+	extends TypedEventEmitter<IContainerEvents>
+	implements DecomposedContainer
+{
+	private _disposed = false; // Track actual disposed state
+
+	public constructor(runtime: IContainerRuntime) {
+		super();
+		this.runtime = runtime;
+		// Note: IContainerRuntime doesn't emit "closed" events like IContainer does
+		// Only bind to events that IContainerRuntime actually supports
+		runtime.on("attached", this.attachedHandler);
+		runtime.on("connected", this.connectedHandler);
+		runtime.on("disconnected", this.disconnectedHandler);
+		runtime.on("dispose", this.disposedHandler);
+	}
+
+	private readonly attachedHandler = (): boolean => this.emit("attached");
+	private readonly connectedHandler = (clientId: string): boolean =>
+		this.emit("connected", clientId);
+	private readonly disconnectedHandler = (): boolean => this.emit("disconnected");
+	private readonly disposedHandler = (): boolean => {
+		this._disposed = true; // Mark as disposed when dispose event occurs
+		// IContainerRuntime emits "dispose" (no error) but we emit "disposed" (optional error) to match IContainerEvents
+		// Since IContainerRuntime doesn't provide error info, we emit without error parameter
+		return this.emit("disposed");
+	};
+
+	private readonly runtime: IContainerRuntime;
+
+	public get audience(): IAudience {
+		return this.runtime.getAudience();
+	}
+
+	public get clientId(): string | undefined {
+		return this.runtime.clientId;
+	}
+
+	public get attachState(): AttachState {
+		return this.runtime.attachState;
+	}
+
+	public get connectionState(): ConnectionState {
+		// TODO: Attempt to map all possible connection states. If the runtime exposes a more granular connectionState property, use it.
+		return this.runtime.connected ? ConnectionState.Connected : ConnectionState.Disconnected;
+	}
+
+	public get closed(): boolean {
+		return this._disposed; // Only return true if actually disposed, not just disconnected
```

Comment:
Nit: might be good to add a bit more detail to this comment. `IContainerRuntime` doesn't have a "closed" state, just "disconnected" (which can still become reconnected) and "disposed".